psi4/driver/p4util/spectrum.py

```diff
@@ -0,0 +1,376 @@
+    gamma : float
+        Broadening parameter.
+        This is the full width at half maximum (FWHM).
+
+    Notes
+    -----
+    Use this profile to model homogeneous broadening.
+    """
+
+    def lineshape(self, x_0: float) -> np.ndarray:
+        """Lorentzian function on `self.domain`, centered at `x_0` with broadening `self.gamma`.
+
+        Parameters
+        ----------
+        x_0 : float
+            Center of the Lorentzian, i.e. its maximum.
+
+        Returns
+        -------
+        lorentz : np.ndarray
+            The Lorentzian profile.
+        """
+        prefactor = 1.0 / np.pi
+        numerator = self.gamma(x_0) / 2.0
+        denominator = (self.domain - x_0)**2 + numerator**2
+
+        return prefactor * (numerator / denominator)
+
+    def maximum(self, x_0: float) -> float:
+        return 2.0 / (np.pi * self.gamma(x_0))
+
+
+def prefactor_opa() -> float:
+    r"""Prefactor for converting microscopic observable to decadic molar
+    extinction coefficient in one-photon absorption.
+
+    Returns
+    -------
+    prefactor : float
+
+    Notes
+    -----
+    This function implements the calculation of the following prefactor:
+
+    .. math::
+
+        k = \frac{4\pi^{2}N_{\mathrm{A}}}{3\times 1000\times \ln(10) (4 \pi \epsilon_{0}) n \hbar c}
+
+    The prefactor is computed in SI units and then adjusted for the fact that
+    we use atomic units to express microscopic observables: excitation energies
+    and transition dipole moments.
+    The refractive index :math:`n` is, in general, frequency-dependent. We
+    assume it to be constant and equal to 1.
+    """
+
+    N_A = qcel.constants.get("Avogadro constant")
+    c = qcel.constants.get("speed of light in vacuum")
+    hbar = qcel.constants.get("Planck constant over 2 pi")
+    e_0 = qcel.constants.get("electric constant")
+    au_to_Coulomb_centimeter = qcel.constants.get("elementary charge") * qcel.constants.get(
+        "Bohr radius") * qcel.constants.conversion_factor("m", "cm")
+
+    numerator = 4.0 * np.pi**2 * N_A
+    denominator = 3 * 1000 * np.log(10) * (4 * np.pi * e_0) * hbar * c
+
+    return (numerator / denominator) * au_to_Coulomb_centimeter**2
+
+
+def prefactor_ecd() -> float:
+    """Prefactor for converting microscopic observable to decadic molar
+    extinction coefficient in electronic circular dichroism.
+
+    Returns
+    -------
+    prefactor : float
+
+    Notes
+    -----
+    This function implements the calculation of the following prefactor:
+
+    .. math::
+
+        k = \frac{16\pi^{2}N_{\mathrm{A}}}{3\times 1000\times \ln(10) (4 \pi \epsilon_{0}) n \hbar c^{2}}
+
+    The prefactor is computed in SI units and then adjusted for the fact that
+    we use atomic units to express microscopic observables: excitation energies
+    and transition dipole moments.
+    The refractive index :math:`n` is, in general, frequency-dependent. We
+    assume it to be constant and equal to 1.
+    """
```

Comment:
Lovely docstrings throughout. I'm looking into if we can get sphinx to pull types from annotations and skip the duplication in docstring.